docs/reference/platforms.rst

```diff
@@ -0,0 +1,218 @@
+Platforms
+=========
+
+Platforms
+---------
+
+The ``platforms`` keyword in a ``charmcraft.yaml`` determines where charms are
+built and where they run.
+
+For more information on how to build charms for specific bases and
+architectures, see the :doc:`Platforms how-to </howto/build-guides/platforms>`
+page.
+
+Standard notation
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on: [<arch>, <arch>]
+       build-for: [<arch>]
+
+The platform name is an arbitrary string that describes the platform. The
+recommended platform name is the ``build-for`` arch.
+
+See :ref:`build-on <reference-build-on>` and
+:ref:`build-for <reference-build-for>` for information on the ``build-on``
+and ``build-for`` keywords.
+
+.. _reference-platforms-shorthand:
+
+Shorthand notation
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <arch>:
+
+Shorthand notation is a simple way to describe charms that build on and build
+for the same architecture. Shorthand notation requires the platform name to be
+a :ref:`valid debian architecture <reference-architectures>`.
+
+.. _reference-platforms-multi-base:
+
+Multi-base notation
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on: [<base>:<arch>, <base>:<arch>]
+       build-for: [<base>:<arch>]
+
+Multi-base charms define a :ref:`base <reference-bases>` in each platform entry
+instead of the top-level ``base`` and ``build-base`` keywords. Within each
+platform entry, the base must be the same for all ``build-on`` and
+``build-for`` entries.
+
+The recommended platform name is ``<distribution>-<series>-<build-for-arch>``
+(for example, ``ubuntu-24.04-riscv64``).
+
+.. _reference-platforms-multi-base-shorthand:
+
+Multi-base shorthand notation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <base>:<arch>:
+
+Multi-base charms can also use shorthand notation when the platform builds on
+and builds for the same architecture. The platform name must be a
+:ref:`valid base <reference-bases>` and
+:ref:`debian architecture <reference-architectures>` formatted as
+``<base>:<arch>``.
+
+.. _reference-build-on:
+
+``build-on``
+~~~~~~~~~~~~
+
+``build-on`` is a list of architectures and optional bases that describes the
+environments where the platform can build. Each entry is formatted as
+``[<base>:]<arch>``.
```

Comment:
This notation twists me in light of `build-for: [<base>:<arch>]` and `<base>:<arch>:`. I don't see any examples where we use it. Are the square brackets literal?

Reply:
Yeah, this is hard to define as-is. I've added an English description of the format and some examples to help disambiguate.

Reply:
I've also converted the examples here to use YAML's multi-line sequence syntax so we shouldn't have any load-bearing brackets

Reply:
Ah, so they're not literal. That's a tricky one to format, since YAML uses square brackets for inline collections. Gah. I've suggested a workaround in my upcoming review.